From bffdb0c398823461000788a60a5ac0db325ee09e Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:17 +0100
Subject: [PATCH 14/52] Revert "Do not unload enum fwd decl but silence redecl
 diag (ROOT-9363 ROOT-9114)."

This reverts commit 6566fa63bc6930550cf373e422aa7825c88d65c6.
---
 .../llvm-project/clang/lib/Sema/SemaDecl.cpp  | 23 -------------------
 1 file changed, 23 deletions(-)

diff --git a/interpreter/llvm-project/clang/lib/Sema/SemaDecl.cpp b/interpreter/llvm-project/clang/lib/Sema/SemaDecl.cpp
index 91add32656..30ae06a323 100644
--- a/interpreter/llvm-project/clang/lib/Sema/SemaDecl.cpp
+++ b/interpreter/llvm-project/clang/lib/Sema/SemaDecl.cpp
@@ -16960,29 +16960,6 @@ bool Sema::CheckEnumRedeclaration(SourceLocation EnumLoc, bool IsScoped,
       return true;
     }
   } else if (IsFixed != Prev->isFixed()) {
-    // Determine whether this is a cling fwd decl.
-    auto hasFwdDeclAnnotation = [](const Decl *Prev) -> bool {
-      for(auto attr = Prev->specific_attr_begin<AnnotateAttr>(),
-               end = Prev->specific_attr_end<AnnotateAttr> ();
-          attr != end;
-          ++attr)
-      {
-        if (!attr->isInherited()) {
-          llvm::StringRef annotation = attr->getAnnotation();
-          assert(!annotation.empty() && "Empty annotation!");
-          static const char annoTag[] = "$clingAutoload$";
-          if (annotation.startswith(llvm::StringRef(annoTag, strlen(annoTag)))) {
-            // autoload annotation.
-            return true;
-          }
-        }
-      }
-      return false;
-    };
-
-    if (hasFwdDeclAnnotation(Prev))
-      return false;
-
     Diag(EnumLoc, diag::err_enum_redeclare_fixed_mismatch)
       << Prev->isFixed();
     Diag(Prev->getLocation(), diag::note_previous_declaration);
-- 
2.43.5

